### DISCUSSION TITLE
Database function does not exist error

### DISCUSSION THREAD

**Discussion by dedok3000** [supabase/supabase#30336] [Questions] [answered]
Hi there,

I've created a database function using the following code in the SQL editor:

```plpgsql
CREATE OR REPLACE FUNCTION public.test_function(data json)
RETURNS json AS $$
BEGIN
    RETURN data;
END;
$$ LANGUAGE plpgsql;
``` 

It shows up in the dashboard and it appears to be visible and accessible when I run the following command:

```plpgsql
SELECT proname, proargtypes FROM pg_proc WHERE proname IN ('test_function');
``` 

However, when I try to send an HTTP POST request to `https://<my-id>.supabase.co/rest/v1/rpc/test_function`

with the following headers:

```
apikey: <my service_role key here>
Authorization: Bearer <my service_role key here>
Content-Type: application/json
```

and `{"test": "data"}` as a payload, I'm getting this response:

`{"code":"42883","details":null,"hint":"No function matches the given name and argument types. You might need to add explicit type casts.","message":"function public.test_function() does not exist"}`

And the API log shows 404 status code...

Why? :(

**Comment by GaryAustin1** [comment, marked as answer]
You need to provide a parameter named data I believe.
See the examples here https://docs.postgrest.org/en/v12/references/api/functions.html for a named parameter of type json and a function that takes unnamed json.

**Reply by dedok3000** [reply]
Oh... yeah looks like when I send `{"data": {"test": "data"}}` it works, thank you.
